config/routes.rb

```diff
@@ -7,6 +7,9 @@
   root "users#index"
 
   resources :users, only: [:index, :show] do 
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do 
+      resources :comments, only: [:create]
```

Comment:
Hi, I would advise you to add `:new` to comment, you need to get this below when you run rails routes
  `POST   /posts/:post_id/comments(.:format)     comments#create`
 `GET   /posts/:post_id/comments/new(.:format)     comments#new`